src/diffusers/schedulers/scheduling_dpmsolver_sde.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2023 Katherine Crowson, The HuggingFace Team and hlky. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torchsde
```

Comment:
As `torchsde` may not be a hard dependency, a possible alternative could be to pass the BrownianTree noise sampler from the outside into the step function, and document this approach in the function's documentation. Do you have any other suggestions on how to proceed?

Reply:
I think we can have `torchsde` as a soft dependency like we have `kdiffusion`. 

@patrickvonplaten WDYT?

Reply:
Agree, think having `torchsde` as a soft dependency here makes sense! The package is not too large and easily installable with 
```
pip install torchsde
```